### ISSUE TITLE
Container orchestration refactor

### ISSUE THREAD

**Issue by w-bonelli** [Computational-Plant-Science/plantit#257]
There's little justification to roll our own system on top of Dask to interface with deployment targets. There's no lack of options in this space, and Dask is better suited to interactive use cases anyway, e.g., in Jupyter notebooks. (It would be nice to provide preconfigured notebooks as tutorials alongside workflows, with examples for how to use the code and parallelize it with Dask, but that's a separate concern.)

[Snakemake](https://snakemake.readthedocs.io/en/stable/index.html) or [NextFlow](https://www.nextflow.io/) do everything we need and more:
- running jobs in containers, including support for Singularity
- integration with cluster schedulers
- environment variables and parametrization

 The [`plantit-cli`](https://github.com/Computational-Plant-Science/plantit-cli) should just expose a handy interface to the `plantit` resource APIs. (Maybe it's worth spinning off the Terrain interface as its own project -- it might be nice to have a CLI to Terrain, although it should probably disallow data transfers beyond a certain size limit to avoid overloading CyVerse web APIs)

**Comment by w-bonelli** [comment]
This is a good opportunity to introduce task composition: check input/output types of each workflow and, if compatible, allow chaining. We should also allow task groups to be scheduled in parallel.

**Comment by w-bonelli** [comment]
Tentatively finished with [36ffc8d](https://github.com/Computational-Plant-Science/plantit/commit/36ffc8d8bff02738a8f19cf77fea1242dcb87775)
